cmake/CompilerWarnings.cmake

```diff
@@ -70,5 +70,6 @@ function(add_more_warnings_to project)
     message(AUTHOR_WARNING "No compiler warnings set for '${CMAKE_CXX_COMPILER_ID}' compiler.")
   endif()
 
-  target_compile_options(${project} INTERFACE ${PROJECT_WARNINGS})
+  # XXX NO! target_compile_options(${project} INTERFACE ${PROJECT_WARNINGS})
+  add_compile_options(${PROJECT_WARNINGS})
```

Comment:
I was trying to prevent the export by hiding them behind the AZMQ developer flag, since Interface targets can't have private compile options, this was the best I could come up with. It never occurred to me to only set it on the directory... I like it.